addons/website/static/src/components/views/theme_preview.xml

```diff
@@ -1,54 +1,77 @@
+        <t t-set="breadcrumb" t-value="breadcrumbs.slice(-1)[0] || {}"/>        
+        <div class="o_breadcrumb d-flex gap-1">
+            <div class="active d-flex flex-column">
+                <h4 class="mb-0"><t t-call="web.Breadcrumb.Name"/></h4>
+            </div>
+        </div>
+    </t>
+
     <t t-name="website.ThemePreviewForm.ControlPanel" t-inherit="web.ControlPanel" t-inherit-mode="primary">
+        <xpath expr="//*[hasclass('o_control_panel')]" position="attributes">
+            <attribute name="class" remove="pb-3" add="pb-2" separator=" "/>
+        </xpath>
+        <xpath expr="//*[hasclass('o_control_panel_main_buttons')]" position="inside">
+            <button class="o_switch_theme btn btn-secondary d-none d-md-inline-block me-1" t-on-click.stop="this.back">
+                <i class="oi oi-arrow-left" role="img" aria-label="Go Back"/>
+            </button>
+        </xpath>
+        <xpath expr="//*[hasclass('o_control_panel_actions')]" position="attributes">
+            <attribute name="class" remove="justify-content-lg-around" add="justify-content-sm-end flex-grow-1 flex-shrink-1 flex-basis-sm-0" separator=" "/>
+        </xpath>
         <xpath expr="//div[hasclass('o_control_panel_navigation')]" position="replace">
-        <div class="d-none d-md-flex align-items-end justify-content-end order-2 order-lg-1 w-100 gap-1" role="group">
-                <input class="btn-check" id="themeViewerDesktop" type="radio" name="viewer" data-mode='desktop' autocomplete="off" checked='checked'/>
-                <label class="btn btn-secondary" for="themeViewerDesktop" t-on-click="this.onDesktopClick">Desktop</label>
-                <input class="btn-check"  id="themeViewerMobile" type="radio" name="viewer" data-mode='mobile' autocomplete="off"/>
-                <label class="btn btn-secondary" for="themeViewerMobile" t-on-click="this.onMobileClick">Mobile</label>
+            <div class="o_control_panel_switch_device d-none d-lg-flex" role="group">
+                <div class="btn-group">
+                    <input class="btn-check" id="themeViewerDesktop" type="radio" name="viewer" data-mode='desktop' autocomplete="off" checked="checked"/>
+                    <label class="btn btn-secondary" for="themeViewerDesktop" t-on-click="this.onDesktopClick">
+                        <i class="fa fa-fw fa-desktop" role="img" aria-label="Desktop"/>
+                    </label>
+                    <input class="btn-check"  id="themeViewerMobile" type="radio" name="viewer" data-mode='mobile' autocomplete="off"/>
+                    <label class="btn btn-secondary" for="themeViewerMobile" t-on-click="this.onMobileClick">
+                        <i class="fa fa-fw fa-mobile" role="img" aria-label="Mobile"/>
+                    </label>
+                </div>
             </div>
         </xpath>
+        <xpath expr="//t[@t-call='web.Breadcrumbs']" position="replace">
```

Comment:
There is a traceback because of this line when we select a theme :+1: 
Doing the xpath like this is not correct => it was recently fixed in [this PR](https://github.com/odoo/odoo/pull/176365/files), so it should be fixed the same way here :wink:

Reply:
Yep, just find out the same! Pushing the update.

Reply:
Green on rb 👍